File: app/src/main/java/com/finallab/smartschoolpickupsystem/FeedbackAdapter.java
package com.finallab.smartschoolpickupsystem;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


import java.text.SimpleDateFormat;
import java.util.List;
import java.util.Locale;

public class FeedbackAdapter extends RecyclerView.Adapter<FeedbackAdapter.FeedbackViewHolder>{

    private final List<feedback> feedbackList;
    private final OnFeedbackLongClickListener longClickListener;

    public interface OnFeedbackLongClickListener {
        void onLongClick(feedback item);
    }

    public FeedbackAdapter(List<feedback> feedbackList, OnFeedbackLongClickListener listener) {
        this.feedbackList = feedbackList;
        this.longClickListener = listener;
    }

    @NonNull
    @Override
    public FeedbackViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.feedback_item, parent, false);
        return new FeedbackViewHolder(view);
    }


    @Override
    public void onBindViewHolder(@NonNull FeedbackViewHolder holder, int position) {
        feedback item = feedbackList.get(position);

        holder.feedbackText.setText("Feedback: " + item.getFeedbackText());
        holder.status.setText("Status: " + item.getStatus());
        holder.adminReply.setText("Reply: " + (item.getAdminReply() == null ? "N/A" : item.getAdminReply()));

        if (item.getTimestamp() != null) {
            String dateStr = new SimpleDateFormat("dd MMM yyyy", Locale.getDefault())
                    .format(item.getTimestamp().toDate());
            holder.timestamp.setText("Date: " + dateStr);
        } else {
            holder.timestamp.setText("Date: N/A");
        }
        // ✅ Enable long-click to trigger delete
        holder.itemView.setOnLongClickListener(v -> {
            longClickListener.onLongClick(item);
            return true;
        });
    }



    @Override
    public int getItemCount() {
        return feedbackList.size();
    }

    public static class FeedbackViewHolder extends RecyclerView.ViewHolder {
        TextView feedbackText, status, adminReply, timestamp;

        public FeedbackViewHolder(@NonNull View itemView) {
            super(itemView);
            feedbackText = itemView.findViewById(R.id.feedbackText);
            status = itemView.findViewById(R.id.feedbackStatus);
            adminReply = itemView.findViewById(R.id.feedbackReply);
            timestamp = itemView.findViewById(R.id.feedbackTimestamp);
        }
    }
}

File: app/src/main/java/com/finallab/smartschoolpickupsystem/Recycler/FeedbackAdapter.kt
package com.finallab.smartschoolpickupsystem.model.Adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.finallab.smartschoolpickupsystem.R
import com.finallab.smartschoolpickupsystem.model.DataModels.Feedback

class FeedbackAdapter(
    private var feedbackList: MutableList<Feedback>,
    private val onStatusUpdate: (Feedback) -> Unit
) : RecyclerView.Adapter<FeedbackAdapter.FeedbackViewHolder>() {

    inner class FeedbackViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val guardianNameText: TextView = view.findViewById(R.id.guardianNameText)
        val feedbackText: TextView = view.findViewById(R.id.feedbackText)
        val sentimentText: TextView = view.findViewById(R.id.sentimentText)
        val statusText: TextView = view.findViewById(R.id.statusText)
        val adminReplyText: TextView = view.findViewById(R.id.adminReplyText)
        val markResolvedBtn: Button = view.findViewById(R.id.markResolvedBtn)
        val statusTick: ImageView = view.findViewById(R.id.statusTick)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FeedbackViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_feedback, parent, false)
        return FeedbackViewHolder(view)
    }

    override fun onBindViewHolder(holder: FeedbackViewHolder, position: Int) {
        val item = feedbackList[position]

        holder.guardianNameText.text = "Guardian: ${item.guardianName}"
        holder.feedbackText.text = "Remarks: ${item.feedbackText}"
        holder.sentimentText.text = "Sentiment: ${item.sentiment}"
        holder.statusText.text = "Status: ${item.status}"

        if (item.status.equals("Resolved", ignoreCase = true)) {
            holder.statusText.setTextColor(holder.itemView.context.getColor(R.color.green))
            holder.statusTick.visibility = View.VISIBLE
            holder.markResolvedBtn.visibility = View.GONE

            if (!item.adminReply.isNullOrBlank()) {
                holder.adminReplyText.visibility = View.VISIBLE
                holder.adminReplyText.text = "Admin Reply: ${item.adminReply}"
            } else {
                holder.adminReplyText.visibility = View.GONE
            }
        } else if (item.status.equals("Pending", ignoreCase = true)) {
            // 🔴 Set status text color to red for pending
            holder.statusText.setTextColor(holder.itemView.context.getColor(R.color.red))
            holder.statusTick.visibility = View.GONE
            holder.markResolvedBtn.visibility = View.VISIBLE
            holder.adminReplyText.visibility = View.GONE

            holder.markResolvedBtn.setOnClickListener {
                onStatusUpdate(item)
            }
        } else {
            // Default fallback if other statuses appear
            holder.statusText.setTextColor(holder.itemView.context.getColor(R.color.black))
            holder.statusTick.visibility = View.GONE
            holder.markResolvedBtn.visibility = View.VISIBLE
            holder.adminReplyText.visibility = View.GONE
        }
    }



    override fun getItemCount(): Int = feedbackList.size

    fun updateList(newList: MutableList<Feedback>) {
        feedbackList = newList
            .sortedWith(compareBy(
                { it.status.equals("Resolved", ignoreCase = true) }, // false (Pending) comes before true (Resolved)
                { it.timestamp } // Optional: sort within groups by time
            ))
            .toMutableList()
        notifyDataSetChanged()
    }

}
